azure-pipelines.yml

```
@@ -77,6 +77,14 @@ steps:
     cargo test --verbose --all --features "test unstable"
   displayName: test
 
+- script: |
+    cargo install --verbose cargo-tarpaulin --version '0.13.3'
+    cargo tarpaulin --verbose --all --features "test unstable" --out Xml
+    bash <(curl -s https://codecov.io/bash)
+  displayName: cov
```

```suggestion
  displayName: code_coverage
  condition: and( eq( variables['Agent.OS'], 'Linux' ), eq( variables['rustup_toolchain'], 'stable' ) )
```

We should limit the code coverage step to Linux on stable so we don't upload more than once. (Note: I'm just guessing the syntax here, so you might need to play with this condition a bit. [Source](https://stackoverflow.com/questions/57411060/azure-pipeline-using-matrix-environment-variable-in-step-condition))